src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -61,6 +61,7 @@ public Type[] GetStartupTypes()
 
         public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
         {
+            // TODO: take a look at this
             string binPath;
             var functionMetadataCollection = _functionMetadataProvider.GetFunctionMetadata(forceRefresh: true);
             HashSet<string> bindingsSet = null;
```

Comment:
It looks like, when extension bundle is configured, we first look at the functions to understand which startup types to load. However, this would be a chicken and egg problem if extensions themselves are indexing the functions.

This is ok if the extension using custom indexer is not part of extension bundle. But when they are, this part may get tricky.